src/bp/ui-studio/src/web/views/FlowBuilder/diagram/nodes/CommentNode.tsx

```diff
@@ -0,0 +1,104 @@
+import classnames from 'classnames'
+import _ from 'lodash'
+import React from 'react'
+import { AbstractNodeFactory, DiagramEngine, NodeModel } from 'storm-react-diagrams'
+
+import { BaseNodeModel } from './BaseNodeModel'
+
+const style = require('./style.scss')
+
+export class CommentNodeWidget extends React.Component<{ node: CommentNodeModel; diagramEngine: DiagramEngine }> {
+  state = {
+    rows: 0,
+    text: ''
+  }
+
+  componentDidMount() {
+    this.setState({ text: this.props.node.text }, this.updateTextareaHeight)
+  }
+
+  handleKeyDown = event => {
+    if (event.key === 'Enter') {
+      this.setState({ text: this.state.text + '\n' }, this.updateTextareaHeight)
+
+      event.preventDefault()
+    }
+  }
+
+  handleOnChange = event => {
+    this.setState({ text: event.target.value }, this.updateTextareaHeight)
+  }
+
+  handleOnBlur = () => {
+    this.props.node.handleTextUpdate(this.state.text)
+  }
+
+  updateTextareaHeight = () => {
+    this.setState({ rows: (_.countBy(this.state.text)['\n'] || 0) + 1 })
+  }
+
+  render() {
+    const node = this.props.node
+
+    const className = classnames(style.comment, { [style.commentSelected]: node.selected })
+
+    return (
+      <div>
+        <textarea
+          className={className}
+          value={this.state.text}
+          onKeyDown={this.handleKeyDown}
+          onChange={this.handleOnChange}
+          onBlur={this.handleOnBlur}
+          rows={this.state.rows}
+        />
+      </div>
```

Comment:
Can you try using BlueprintJs TextArea instead of the standard one ? I think it would be able to handle automatically its size (so no need of the logic for height)

https://blueprintjs.com/docs/#core/components/text-inputs.text-area

Reply:
Well, `TextArea` can automatically grow, but I couldn't find a way to automatically make it "shrink", when the user removes lines, so we need to calculate `rows` attribute anyway.

A border was added, please beautify it as you wish.

I wonder how we can prevent dragging inside the `TextArea`?